pkg/horoscope/card.go

```diff
@@ -0,0 +1,182 @@
+package horoscope
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/chaos-mesh/horoscope/pkg/executor"
+	log "github.com/sirupsen/logrus"
+)
+
+type CardinalityQueryType int
+
+const (
+	TypeEMQ CardinalityQueryType = iota
+	TypeRGE
+	TypeDCT
+)
+
+type Cardinalitor struct {
+	exec         executor.Executor
+	Type         CardinalityQueryType
+	TableColumns map[string][]string
+	Timeout      *time.Duration
```

Comment:
Shall we just use `time.Duration` instead of it's pointer？ As the underlying type of `time.Duration` is `int64`.

Reply:
Resolved on 2c687c5e3b7c5e0a846415800db511f5aa376a52